packages/workflow/src/errors/abstract/node.error.ts

```diff
@@ -45,6 +45,8 @@ export abstract class NodeError extends ExecutionBaseError {
 		}
 
 		this.node = node;
+
+		if (error instanceof NodeError) return error;
```

Comment:
If we know `error` is already a successfully constructed `NodeError`, there's no need for the extra work above? Can we check and return it immediately at the top of the constructor?

Reply:
I couldn't get that to work before of TS errors like these.
![image](https://github.com/n8n-io/n8n/assets/196144/3964e79a-7e7f-4043-9d9b-04c1bba00d08)

Reply:
okay, managed to get it to work, but had to move the `super` call out of the `if` to make tsc happy.